Incorrect compiled output for external variant references inside match patterns

The following code does compile, but does not produce the expected result and instead results in Go code that does not run:
```rs
import "go:bufio"
import "go:fmt"
import "go:io"
import "go:os"

fn main() {
  let reader = bufio.NewReader(os.Stdin)
  while true {
    let r = match reader.ReadRune() {
      Ok((r, _)) => r,
      Err(io.EOF) => break, // this line is the issue
      Err(_) => panic("Error"),
    }
    fmt.Printf("%c", r)
  }
}
```
When run it results in the following error:
> ./main.go:26:24: subject_3.ErrVal.Tag undefined (type error has no field or method Tag)
./main.go:26:39: undefined: lisette.errorEOF

That's because the code that was generated for that line with `io.EOF` on it is this:
```go
if subject_3.ErrVal.Tag == lisette.errorEOF {
  break loop_1
}
```
Which is invalid Go.

I can change the line that fails to `Err(err) if err == io.EOF => break,` and everything works as expected, but I would expect the above code to work also, or at least give an error from the lisette compiler/lsp, not fail when Go tries to run it.

--- a/crates/emit/src/go/patterns/decision_tree.rs
+++ b/crates/emit/src/go/patterns/decision_tree.rs
@@ -712,6 +712,14 @@ fn collect_enum_variant_checks(
         return;
     }
 
+    if emitter.as_enum(ty).is_none() && identifier.contains('.') {
+        collector.checks.push(Check::Literal {
+            path: path.clone(),
+            go_literal: identifier.to_string(),
+        });
+        return;
+    }
+
     collect_tagged_enum_checks(emitter, path, &variant_data, collector);
 }
 

--- a/tests/spec/emit/interop_matrix.rs
+++ b/tests/spec/emit/interop_matrix.rs
@@ -1033,3 +1033,26 @@ fn main() {
 "#;
     assert_emit_snapshot!(input);
 }
+
+#[test]
+fn interop_err_sentinel_pattern() {
+    let input = r#"
+import "go:bufio"
+import "go:fmt"
+import "go:io"
+import "go:os"
+
+fn main() {
+  let reader = bufio.NewReader(os.Stdin)
+  while true {
+    let r = match reader.ReadRune() {
+      Ok((r, _)) => r,
+      Err(io.EOF) => break,
+      Err(_) => panic("error"),
+    }
+    fmt.Printf("%c", r)
+  }
+}
+"#;
+    assert_emit_snapshot!(input);
+}

